rstudio/rstudio
# Add "rename" command to document tab context menu

### Intent

Ability to rename a file by right-clicking on its tab is a common feature in IDEs.

Fixes #8374

### Approach

Added rename behavior to document tab context menu.

Can't directly use the rename command itself because it assumes the target is the currently selected editor which isn't necessarily the case when you right-click on a tab.

### QA Notes

Verify that "Rename" only appears when you right-click on tabs of actual saved documents, not on "Untitled" docs or non-document tabs. Behavior should be identical (for better or worse) with the existing File / Rename command on main menu.
